### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by XanderStrike**
Nbd.

### CODE REVIEW

**Comment by bboe** [comment]
Awesome.

File: README.md [hunk 1]
```diff
@@ -9,8 +9,39 @@ _rentlinx_client_ is a ruby wrapper for the RentLinx API.
...
+
+    prop = Property.new(attributes)
+    prop.post # creates
+
+    prop.website = 'http://example.com'
+    prop.post # updates
+
+    prop = Property.from_id('property-id') # fetches
+
+Objects can be created for all endpoints supported by the Rentlinx API, including Units, Ammenities, Photos, and Links. For more detailed information, see the [full documentation on rubygems](http://www.rubydoc.info/gems/rentlinx).
```
Comment (scuendet) [inline README.md]: Amenities, not Ammenities

File: lib/rentlinx.rb [hunk 2]
```diff
@@ -29,6 +42,7 @@ def username(*args)
       end
     end
 
+    # Sets and retrieves the password used ot log in to Rentlinx
```
Comment (scuendet) [inline lib/rentlinx.rb]: to no ot

File: lib/rentlinx/client.rb [hunk 3]
```diff
@@ -16,6 +21,10 @@ class Client
     include Rentlinx::AmenityClientMethods
     include Rentlinx::LinkClientMethods
 
+    # Returns a new instance of client. Always use Rentlinx.client to
+    # avoid making unnecessary extra connections to Rentlinx.
```
Comment (scuendet) [inline lib/rentlinx/client.rb]: what does this mean? "Always use Rentlinx.client to... " : this is not clear to me, what do you mean?

File: lib/rentlinx/errors.rb [hunk 4]
```diff
@@ -1,31 +1,50 @@
 module Rentlinx
+  # Basic error class to be inherited from
+  #
+  # Use this error to rescue all types of Rentlinx errors
+  #
+  # @example
```
Comment (scuendet) [inline lib/rentlinx/errors.rb]: is that example really needed?

File: lib/rentlinx/models/property.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@
 require 'rentlinx/modules/linkable'
 
 module Rentlinx
+  # An object that represents a Rentlinx property.
```
Comment (scuendet) [inline lib/rentlinx/models/property.rb:6]: detail, but seems like sometimes you end sentences with a period, and sometimes you don't

File: lib/rentlinx/models/property.rb [hunk 6]
```diff
@@ -15,15 +16,26 @@ class Property < Base
 
     attr_accessor(*ATTRIBUTES)
 
+    # Posts the property with all of its attached units
```
Comment (scuendet) [inline lib/rentlinx/models/property.rb]: I thought we wanted to get rid of this type of methods that embed the posting of something else with it. If you post a unit here, do you expect it to post with all its component, (amenities, etc.), or just the units?

If we agree to remove these, we could probably do it now.

File: lib/rentlinx/models/property.rb [hunk 7]
```diff
@@ -31,21 +43,31 @@ def units=(unit_list)
       end
     end
 
+    # Querys rentlinx and builds a property with the given ID
```
Comment (scuendet) [inline lib/rentlinx/models/property.rb]: :s/Querys/Queries

File: lib/rentlinx/models/unit.rb [hunk 8]
```diff
@@ -16,24 +17,37 @@ class Unit < Base
...
     end
 
+    # Gets all the photos for a unit.
+    #
+    # @return a list of Rentlinx::UnitPhoto objects
     def photos
       super.select { |p| p.unitID == unitID }
     end
 
+    # Querys rentlinx and builds a unit with the given ID
```
Comment (scuendet) [inline lib/rentlinx/models/unit.rb]: same as above

File: lib/rentlinx/modules/amenityable.rb [hunk 9]
```diff
@@ -1,10 +1,16 @@
 module Rentlinx
+  # A module that encapsulates all ammenity related logic for {Rentlinx::Property} and
+  # {Rentlinx::Unit} objects. All of these methods can be called on Properties and Units.
+  #
+  # TODO: Refactor into BaseAble class along with {Linkable} and {Photoable}
   module Amenityable
+    # Posts the object with its associated amenities
     def post_with_amenities
```
Comment (scuendet) [inline lib/rentlinx/modules/amenityable.rb:11]: again, I know you didn't change that, but I think we should get rid of this method

File: lib/rentlinx/modules/amenityable.rb [hunk 10]
```diff
@@ -34,6 +47,13 @@ def amenities=(amenity_list)
       end
     end
 
+    # Builds a single amenity and attaches it to the object. This method
+    # conveniently assigns the property and unitID (if present) of the
```
Comment (scuendet) [inline lib/rentlinx/modules/amenityable.rb]: s/property/propertyID

File: lib/rentlinx/modules/linkable.rb [hunk 11]
```diff
@@ -1,10 +1,16 @@
 module Rentlinx
+  # A module that encapsulates all link related logic for {Rentlinx::Property} and
+  # {Rentlinx::Unit} objects. All of these methods can be called on Properties and Units.
+  #
+  # TODO: Refactor into BaseAble class along with {Amenityable} and {Photoable}
   module Linkable
+    # Posts the object with associated links
     def post_with_links
```
Comment (scuendet) [inline lib/rentlinx/modules/linkable.rb:11]: same as above, should be remove

**Comment by scuendet** [comment]
Thanks for doing this, Alex! This looks good, just left some comments.

File: lib/rentlinx/models/property.rb (hunk 6 above)
Comment (XanderStrike) [inline lib/rentlinx/models/property.rb]: I'll put a todo, but I'm trying not to change functionality with this branch.

**Pull request merged**